test/e2e/specs/shared-blocks.test.js

```diff
@@ -0,0 +1,207 @@
+		// Insert a paragraph block
+		await insertBlock( 'Paragraph' );
+		await page.keyboard.type( 'Hello there!' );
+
+		// Trigger isTyping = false
+		await page.mouse.move( 200, 300 );
+		await page.mouse.move( 250, 350 );
+
+		// Convert block to a shared block
+		await page.click( 'button[aria-label="More Options"]' );
+		const convertButton = await page.waitForXPath( '//button[text()="Convert to Shared Block"]' );
+		await convertButton.click();
+
+		// Wait for creation to finish
+		await page.waitForXPath(
+			'//*[contains(@class, "notice-success")]/*[text()="Block created."]'
+		);
+
+		// Select all of the text in the title field by triple-clicking on it. We
+		// triple-click because, on Mac, Mod+A doesn't work. This step can be removed
+		// when https://github.com/WordPress/gutenberg/issues/7972 is fixed
+		await page.click( '.shared-block-edit-panel__title', { clickCount: 3 } );
+
+		// Give the shared block a title
+		await page.keyboard.type( 'Greeting block' );
+
+		// Save the shared block
+		const [ saveButton ] = await page.$x( '//button[text()="Save"]' );
+		await saveButton.click();
+
+		// Wait for saving to finish
+		await page.waitForXPath( '//button[text()="Edit"]' );
+
+		// Check that we have a shared block on the page
+		const block = await page.$( '.editor-block-list__block[data-type="core/block"]' );
+		expect( block ).not.toBeNull();
+
+		// Check that its title is displayed
+		const title = await page.$eval(
+			'.shared-block-edit-panel__info',
+			( element ) => element.innerText
+		);
+		expect( title ).toBe( 'Greeting block' );
+	} );
+
+	it( 'can be created with no title', async () => {
+		// Insert a paragraph block
+		await insertBlock( 'Paragraph' );
+		await page.keyboard.type( 'Hello there!' );
+
+		// Trigger isTyping = false
+		await page.mouse.move( 200, 300 );
+		await page.mouse.move( 250, 350 );
+
+		// Convert block to a shared block
+		await page.click( 'button[aria-label="More Options"]' );
+		const convertButton = await page.waitForXPath( '//button[text()="Convert to Shared Block"]' );
+		await convertButton.click();
+
+		// Wait for creation to finish
+		await page.waitForXPath(
+			'//*[contains(@class, "notice-success")]/*[text()="Block created."]'
+		);
+
+		// Save the shared block
+		const [ saveButton ] = await page.$x( '//button[text()="Save"]' );
+		await saveButton.click();
+
+		// Wait for saving to finish
+		await page.waitForXPath( '//button[text()="Edit"]' );
+
+		// Check that we have a shared block on the page
+		const block = await page.$( '.editor-block-list__block[data-type="core/block"]' );
+		expect( block ).not.toBeNull();
+
+		// Check that it is untitled
+		const title = await page.$eval(
+			'.shared-block-edit-panel__info',
+			( element ) => element.innerText
+		);
+		expect( title ).toBe( 'Untitled shared block' );
+	} );
+
+	it( 'can be inserted and edited', async () => {
+		// Insert the shared block we created above
+		await insertBlock( 'Greeting block' );
```

Comment:
There should ideally be no concept of "above" as assuming tests to be run in particular order (or even that any others are run at all). I should be able to change this to [`it.only`](https://jestjs.io/docs/en/api#testonlyname-fn-timeout) and the test should be fine to run in isolation. Common behaviors should be established in lifecycle (`beforeEach` or `beforeAll`) or repeated in every test case.

Note: I'm encountering this in implementing a new end-to-end test case for verifying the fix at #21205. Ideally I'd like to iterate on the test case with `it.only`, but at least as currently presented, there's no prior art for me to leverage commonly available reusable blocks to work with. For the purposes of my test, I might just copy the logic from "can be created" to create the reusable block. It could be straight-forward enough to refactor to its own function; not sure if I'll do that or end up just following the pattern here of reusing the block created in an earlier test case once finished.

Reply:
Yes, you are right, the setup code belongs in a `beforeAll()`. I think this might have been the first E2E test I ever wrote. _Ah, to be young..._

I created https://github.com/WordPress/gutenberg/issues/21249 to track this.